pkg/control/pubcontrol/pub_control.go

```diff
@@ -92,14 +93,17 @@ func (c *commonControl) GetPodsForPub(pub *policyv1alpha1.PodUnavailableBudget)
 	if err = c.List(context.TODO(), podList, listOptions, utilclient.DisableDeepCopy); err != nil {
 		return nil, 0, err
 	}
-
 	matchedPods := make([]*corev1.Pod, 0, len(podList.Items))
 	for i := range podList.Items {
 		pod := &podList.Items[i]
 		if kubecontroller.IsPodActive(pod) {
 			matchedPods = append(matchedPods, pod)
 		}
 	}
+	if pub.Annotations[policyv1alpha1.PubProtectTotalReplicas] != "" {
+		expectedCount, _ := strconv.ParseInt(pub.Annotations[policyv1alpha1.PubProtectTotalReplicas], 10, 32)
```

Comment:
shall we check workload targetRef first, if targetRef is nil , then use the annotation?